### PR TITLE
add several primitive ops

### PR DESCRIPTION
**Pull request by wenming2014**
primitive op demo: replace batch norm with primitive ops
add const_scalar and broadcast_to primitive ops

### CODE REVIEW

File: cinn/frontend/syntax.h [hunk 1]
```diff
@@ -159,6 +159,9 @@ struct Instruction : public common::Shared<_Instruction_> {
 struct Program {
   using attr_t = hlir::framework::NodeAttr::attr_t;
   void SetInputs(const std::vector<Variable>& xs);
+
+  template <typename PrimType>
+  Variable primitive_const_scalar(PrimType value, const std::string& name);
```
Comment (haozech) [inline cinn/frontend/syntax.h:178]: 这里建议加一下注释

File: cinn/frontend/syntax.h [hunk 2]
```diff
@@ -234,6 +308,13 @@ struct Program {
                      const Variable& variance,
                      const std::unordered_map<std::string, attr_t>& attr_store);
 
+  Variable fused_batchnorm_inference(const Variable& a,
```
Comment (haozech) [inline cinn/frontend/syntax.h:328]: 同上，加一下注释

File: cinn/hlir/pass/primitive_ops_demo.cc [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  Placeholder Mean(Float(32), {64}, "Mean");
+  Placeholder Variance(Float(32), {64}, "Variance");
+
+  Program program;
+  std::unordered_map<std::string, Program::attr_t> attrs;
+  attrs["epsilon"] = static_cast<float>(0.001);
+
+  auto a = program.batchnorm(A, Scale, Bias, Mean, Variance, attrs);
+
+  auto b = program.fused_batchnorm_inference(A, Scale, Bias, Mean, Variance, attrs);
```
Comment (haozech) [inline cinn/hlir/pass/primitive_ops_demo.cc]: a和b这两个tensor是什么关系呢？(batchnorm和fused_batchnorm_inference)

File: cinn/hlir/pass/primitive_ops_demo.cc [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  hlir::framework::ApplyPass(graph.get(), "AlterLayout");
+#endif
+  hlir::framework::ApplyPass(graph.get(), "OpFusion");
+  auto scope = BuildScope(target, graph);
+  LOG(INFO) << "graph:\n" << graph->Visualize();
+
+  hlir::framework::GraphCompiler gc(target, scope, graph);
+  auto runtime_program = gc.Build();
+
+  scope->Var<hlir::framework::Tensor>("A");
```
Comment (haozech) [inline cinn/hlir/pass/primitive_ops_demo.cc]: 这里input tensor A的name不是定义的`"C"`么，这里是否应该改为`"C"`

File: cinn/hlir/pass/primitive_ops_demo.cc (hunk 3 above)
Comment (wenming2014) [reply cinn/hlir/pass/primitive_ops_demo.cc]: 2者功能相同，fused_batchnorm_inference是由元算子组合构成

**Review by SunNy820828449 — DISMISSED** [review]
LGTM

File: cinn/hlir/framework/graph_compiler.cc [hunk 5]
```diff
@@ -118,6 +118,18 @@ std::vector<ir::LoweredFunc> GraphCompiler::GetOpFunc(const Node* node) {
   return func;
 }
 
+int GetMasterRefNode(const std::vector<Node*>& nodes) {
```
Comment (Superjomn) [inline cinn/hlir/framework/graph_compiler.cc:122]: add some comments here

**Review by Superjomn — DISMISSED** [review]
LGTM

**Review by haozech — APPROVED** [review]
LGTM
